app/gui/src/dashboard/hooks/backendHooks.ts

```diff
@@ -359,6 +360,65 @@ export function listDirectoryQueryOptions(options: ListDirectoryQueryOptions) {
   })
 }
 
+/**
+ * Options for {@link assetFromCacheQueryOptions}.
+ */
+export interface AssetFromCacheQueryOptions {
+  readonly backend: Backend
+  readonly assetId: AssetId
+  readonly queryClient: QueryClient
+}
+
+/**
+ * Build a query options object to fetch an asset from the React Query cache.
+ * This is _only_ for situations when WE KNOW that the asset is in the cache.
+ * This is _not_ a general purpose function for fetching assets.
```

Comment:
i recommend marking as `@deprecated` to make it more obvious

Reply:
It’s not deprecated; it’s unsafe in some way - anyone using this must be absolutely certain of their actions.

Reply:
yeah, agree on making it `unsafe`